pil_analyzer/src/pil_analyzer.rs

```diff
@@ -489,7 +489,12 @@ impl<T: FieldElement> PILAnalyzer<T> {
     }
 
     fn namespaced_ref(&self, namespace: &Option<String>, name: &str) -> String {
-        format!("{}.{name}", namespace.as_ref().unwrap_or(&self.namespace))
+        if name.starts_with('%') || (namespace.is_none() && self.constants.contains_key(name)) {
```

Comment:
In the case where `name` starts with `%`, seems like `namespace` should be `None`? If so, it would be good to assert that.

Reply:
Added.